src/leader.py

```diff
@@ -152,6 +163,62 @@ def lead(self):
             self.turn(code, position)
 
 
+    def demo_circle(self):
+        """
+        Controls the leader vehicle in a circle for demo purposes.
+
+        Drives the leader vehicle in a continuous circle till script is
+        terminated.
+        """
+        while True:
```

Comment:
You don't need a while loop here since the entire system is already running one

Reply:
If you called this method without a while loop it would set the front wheels to left turn, set the speed to demo speed, and set back wheels to forward. Then immediately set the car back to the current controller code , position for speed, angle and movement. this allows for testing and some sort of exit statement to be added replacing the while True: statement

Reply:
Isn't that supposed to be fixed in the demo mapping though (Zein's branch). This loop never exits this function, which breaks the whole functionality of the system.